lib/auth/tun.go

```diff
@@ -590,8 +597,19 @@ func (c *TunClient) Close() error {
 
 // GetDialer returns dialer that will connect to auth server API
 func (c *TunClient) GetDialer() AccessPointDialer {
-	return func() (net.Conn, error) {
-		return c.Dial(c.authServers[0].AddrNetwork, "accesspoint:0")
+	addrNetwork := c.staticAuthServers[0].AddrNetwork
+	const dialRetryTimes = 5
+
+	return func() (conn net.Conn, err error) {
+		for attempt := 0; attempt < dialRetryTimes; attempt++ {
+			conn, err = c.Dial(addrNetwork, "accesspoint:0")
+			if err == nil {
+				return conn, nil
+			}
+			time.Sleep(dialRetryInterval * time.Duration(attempt))
```

Comment:
instead of `time.Sleep` I recommend using ticker to do the following:

```go
select {
    case <-ticker.C:
    case <-closeC:// this will let you to immediately react on closing of the client, right now groutine is unaware of the fact that the client has closed because of that.